git-secrets-scan-action/entrypoint.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+set -e
+
+# Clone and install git-secrets from source
+cd ${HOME}
+git clone https://github.com/awslabs/git-secrets.git && cd git-secrets && make install 
+
+# Check if GITHUB_WORKSPACE is set and points to a valid directory
+if [[ -z "${GITHUB_WORKSPACE}" || ! -d "${GITHUB_WORKSPACE}" ]]; then
+    echo "Required env variable GITHUB_WORKSPACE not set or does not point to a valid directory."
+    exit 1
+fi
+
+cd ${GITHUB_WORKSPACE}
+
+# Check if repository is checked-out using actions/checkout
+if [ -z "$(ls -A .)" ]; then
+    echo "Workspace is empty. Did you checkout the repository using actions/checkout?"
```

Comment:
I don't think the directory we're checking has to come from `actions/checkout`.

Reply:
Per https://github.com/actions/checkout#Checkout-multiple-repos-side-by-side, any checkout action occurs in `GITHUB_WORKSPACE` by default. (or `GITHUB_WORKSPACE/path` if a path is provided in actions/checkout).

Reply:
Added `path` input variable to accommodate the repository being checked out at a different location within `GITHUB_WORKSPACE`